frontend/src/services/notebookService.ts

```diff
@@ -290,24 +290,32 @@ export const startNotebook = (data: StartNotebookData): Promise<Notebook> => {
   });
 };
 
-export const stopNotebook = (projectName: string, notebookName: string): Promise<Notebook> => {
+export const stopNotebook = async (
+  projectName: string,
+  notebookName: string,
+): Promise<Notebook> => {
   const dateStr = new Date().toISOString().replace(/\.\d{3}Z/i, 'Z');
-  const patch: RecursivePartial<Notebook> = {
+  const notebook = await getNotebook(projectName, notebookName);
+  const patch: Notebook = _.merge({}, notebook, {
     metadata: { annotations: { 'kubeflow-resource-stopped': dateStr } },
-  };
-
-  return patchNotebook(projectName, notebookName, patch);
+  });
+  return replaceNotebook(projectName, notebookName, patch);
 };
 
-export const patchNotebook = (
+export const replaceNotebook = async (
   projectName: string,
   notebookName: string,
-  updateData: RecursivePartial<Notebook>,
+  data: Notebook,
 ): Promise<Notebook> => {
   const url = `/api/notebooks/${projectName}/${notebookName}`;
+  const notebook = _.cloneDeep(await getNotebook(projectName, notebookName));
+  
+  notebook.spec = data.spec;
+  notebook.metadata.annotations['kubeflow-resource-stopped'] =
+    data.metadata.annotations['kubeflow-resource-stopped'];
```

Comment:
So we do have annotations and labels in the notebook... although we don't need to probably carry over `labels` in its current implementation, it'll just generate a bug if we don't.

Copy whole hog the annotations & labels -- hopefully this works as intended. Annotations cover everything from `kubeflow-resource-stopped` to the last selection values they used in the form.

Reply:
Okay will copy but maybe the merge would do it too in this case no?

Reply:
Hmm, not sure what the old code looked like, but yeah probably. I must have missed it if that's the case.